pkg/cmd/deploy/requests.go

```diff
@@ -104,36 +107,87 @@ func (cmd *DeployCmd) doDomain(client *apidom.Client, ctx context.Context, conf
 	return domainReturnedName[0], nil
 }
 
-func (cmd *DeployCmd) doOrigin(client *apiapp.Client, clientorigin *apiori.Client, ctx context.Context, conf *contracts.AzionApplicationOptions) error {
-	if conf.Origin.Id == 0 {
-		err := cmd.createAppRequirements(client, clientorigin, ctx, conf)
+func (cmd *DeployCmd) doOrigin(client *apiapp.Client, clientOrigin *apiori.Client, ctx context.Context, conf *contracts.AzionApplicationOptions, mode string) error {
+	var addresses []string
+	var DefaultOrigin = [1]string{"httpbin.org"}
+
+	if conf.Origin.SingleOriginID == 0 {
+		reqSingleOrigin := apiori.CreateRequest{}
+
+		if len(conf.Origin.Address) > 0 {
+			address := prepareAddresses(conf.Origin.Address)
+			reqSingleOrigin.SetAddresses(address)
+		} else {
+			addresses := prepareAddresses(DefaultOrigin[:])
+			reqSingleOrigin.SetAddresses(addresses)
+		}
+
+		reqSingleOrigin.SetName(utils.Concat(conf.Name, "_single"))
+		reqSingleOrigin.SetHostHeader("${host}")
+
+		origin, err := clientOrigin.Create(ctx, conf.Application.ID, &reqSingleOrigin)
 		if err != nil {
+			logger.Debug("Error created object storage origin", zap.Any("Error", err))
```

Comment:
Error while creating origin of type object storage

Reply:
Wouldn't that be better? Error creating default origin